pkg/credentialprovider/aws/aws_credentials.go

```diff
@@ -66,9 +65,9 @@ type ecrProvider struct {
 	getter tokenGetter
 }
 
-// init registers the various means by which ECR credentials may
+// Init registers the various means by which ECR credentials may
 // be resolved.
-func init() {
+func Init() {
```

Comment:
Add a comment above saying something like "// Not using the package init() function: this module should be initialized only if using the AWS cloud provider.  This avoids timeout waiting for non-existent provider."